NAudio/Wave/WaveInputs/WasapiCapture.cs

```diff
@@ -224,28 +228,54 @@ public void StartRecording()
         /// </summary>
         public void StopRecording()
         {
-            if (captureState != CaptureState.Stopped)
-                captureState = CaptureState.Stopping;
+            // Set Stopping to CaptureState if the value is not Stopped
+            while (true)

```

Comment:
the idea behind the stop recording method is that it only requests that recording stops. It does not block until stopping has complete. This change looks like it will max out CPU until current state gets set to stopped.

Reply:
It does not wait for stopping. It assigns `Stopping` to `currentState` using `CompareExchange` pattern to prevent from overwriting the field when the recoding thread writes to `currentState` at the same time.